### PR TITLE
[node-downloader-helper] Upgrade missing methods and events

### PR DESCRIPTION
**Pull request by robertoaraujom**
Upgrade missing methods, events, options and interfaces

### CODE REVIEW

File: types/node-downloader-helper/index.d.ts [hunk 1]
```diff
@@ -39,7 +62,11 @@ export enum DH_STATES {
 }
 
 export interface DownloaderHelper {
-    on(event: 'start' | 'download' | 'end' | 'pause' | 'resume' | 'stop' | string, callback: () => void): this;
+    on(event: 'start' | 'timeout' | 'pause' | 'stop' | string, callback: () => void): this;
+    on(event: 'download' | 'end', callback: (downloadInfo: DownloadInfo) => void): this;
```
Comment (RemyJeancolas) [inline types/node-downloader-helper/index.d.ts:66]: According to the module's README, it looks like the `DownloadInfo` objects from the **download** and the **end** events are slightly different.

The `DownloadInfo` objet for a **download** event has a `isResumed` property that doesn't exist for the **end** event, while the **end** event has a `onDiskSize` property that doesn't exist in the **download** event.

These differences could induce uncaught errors during development if not properly added to the typescript definitions file.

**Review by RemyJeancolas — CHANGES_REQUESTED** [review]
Hi @Roberto-Araujo , it seems there are some errors in your typings definitions file when comparing with the official module's documentation.

**Comment by Roberto-Araujo** [comment]
Hi @RemyJeancolas, I reviewed official module's documentation, fixed mentioned errors, found that 'error' event callback had an incorrect typing definition.

**Review by RemyJeancolas — APPROVED** [review]
Looks good to me

**Pull request merged**